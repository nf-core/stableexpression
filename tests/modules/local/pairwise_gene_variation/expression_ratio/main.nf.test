nextflow_process {

    name "Test Process EXPRESSION_RATIO"
    script "modules/local/pairwise_gene_variation/expression_ratio/main.nf"
    process "EXPRESSION_RATIO"
    tag "expression_ratio"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                file = file( '$projectDir/tests/test_data/pairwise_gene_variation/cross_join/output/cross_join.0.1.parquet', checkIfExists: true)
                input[0] = file
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
