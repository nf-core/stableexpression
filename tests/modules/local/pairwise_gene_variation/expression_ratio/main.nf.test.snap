{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "ratios.0.1.parquet:md5,b0999933fae92eab5ba6e01a17a352df"
                ],
                "1": [
                    [
                        "EXPRESSION_RATIO",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSION_RATIO",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "data": [
                    "ratios.0.1.parquet:md5,b0999933fae92eab5ba6e01a17a352df"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-29T15:30:55.228354841"
    }
}